internal/statesync/reactor.go

```diff
@@ -102,6 +103,15 @@ const (
 	maxLightBlockRequestRetries = 20
 )
 
+//go:generate mockery --case underscore --name SyncReactor
+// create this interface for testing abilities of node.startStateSync
+type SyncReactor interface {
+	Sync(context.Context, time.Duration) (sm.State, error)
+	Backfill(sm.State) error
+	GetLogger() log.Logger
```

Comment:
We shouldn't need this method.

Reply:
it's for the TestNodeStartStateSync, the mock instance is not able to retrieve Logger directly.

Reply:
Actually, I'm not quite sure to create this interface for the testing purpose is good. But I couldn't find a good way to test the function in `node` struct.  I can remove all the interfaces and the test if you think keeping the original `node` and `statesync reactor` structure is better.

Reply:
Could it be easier if you just pass the event bus into the statesync.Sync() function and use it there?

Reply:
@cmwaters right, but the logger module will be different, will it make some confusion?

Reply:
We pass the event bus to the consensus reactor and log from within there for example